test/gpu.jl

```diff
@@ -49,7 +50,63 @@ end
     @test norm(scfres_cpu.ρ - Array(scfres_gpu.ρ)) < 1e-6
 end
 
+@testset "CUDA Magnetic+other terms" begin
```

Comment:
That's a lot of code for a test of a rather obscure feature. Can you not just test the hamiltonian apply of the terms or something like that? See the hamiltonian_consistency.jl test if you're unsure what I mean. 

Maybe you can easily add a gpu branch to it ... if you come up with some reasonable base functions and stick them into some common file it should not imply too much extra code.

Reply:
Good point, I have now added dedicated "hamiltonian_consistency" tests for GPUs, although this requires lifting the common functions to a different source file.